### PR TITLE
[exporter/googlecloud] Update GCP dependencies (v0.35.0/v1.11.0)

### LINKED ISSUE #18197: [exporter/googlecloud] Bump to v1.11.0/v0.35.0
### Component(s)

exporter/googlecloud
exporter/googlemanagedprometheus

### Describe the issue you're reporting

Tracking issue for the GCP exporter bump (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18182), full release notes at https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/releases/tag/v0.35.0

Also relevant is the GMP exporter being marked beta (included in the same PR and release notes). This includes updates to be specification compliant, such as reporting `target_info` and `scope_info` metrics.

### PR DESCRIPTION
**Pull request by damemi**
**Description:**
This updates the GCP dependencies to the latest release https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/releases/tag/v0.35.0

As part of that, this also promotes the GMP exporter to beta stability.

**Link to tracking Issue:** N/A

**Testing:** Downstream integration and unit tests

**Documentation:** README updated here and downstream

Fixes #18197

### CODE REVIEW

**Comment by damemi** [comment]
How should we handle the required issue in changelog entries? We have downstream issues in our repo that I can link

File: .chloggen/gcp-v0350-update.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: googlecloudexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: New release of GCP exporter
+
+# One or more tracking issues related to the change
+issues: []
```
Comment (dmitryax) [inline .chloggen/gcp-v0350-update.yaml:11]: Please update this field

**Comment by dashpole** [comment]
@damemi feel free to open a tracking issue here as well

File: .chloggen/gcp-v0350-update.yaml (hunk 1 above)
Comment (damemi) [reply .chloggen/gcp-v0350-update.yaml]: Good to know, that wasn't clear. Opened a tracking issue instead

**Review by dashpole — APPROVED** [review]

**Pull request merged**